File: Contracts/Model/Product/UserInteract/Response/PurchaseDetailResponse.cs
using Contracts.Model.Product.Response;

namespace Contracts.Model.Product.UserInteract.Response;

public class PurchaseDetailResponse : BaseResponse
{
	public Guid Id { get; set; }
	public Guid ProductId { get; set; }
	public ICollection<PurchaseDetailOptionResponse> PurchaseOptions { get; set; }
	public Guid? OrderId { get; set; }
	public Guid? QuoteId { get; set; }
	public Guid? WarrantyId { get; set; }
	public WarrantyPreviewResponse? Warranty { get; set; }
}

File: ClientApp/src/app/core/interfaces/Product/UserInteract/Response/purchaseDetailResponse.ts
import {BaseResponse} from "../../../baseResponse";
import {PurchaseDetailOptionResponse} from "./purchaseDetailOptionResponse";
import {WarrantyPreviewResponse} from "@core/interfaces/Product/Response/warrantyPreviewResponse";

export interface PurchaseDetailResponse extends BaseResponse {
  purchaseNumber: string;
  productId: string;
  purchaseOptions: PurchaseDetailOptionResponse[];
  cartId: string;
  orderId: string | null;
  quoteId: string | null;
  warrantyId: string | null;
  warranty: WarrantyPreviewResponse | null;
  category: string;
}
